pyatlan/model/search.py

```diff
@@ -1848,6 +1849,11 @@ class IndexSearchRequest(SearchRequest):
         alias="allowDeletedRelations",
     )
 
+    def get_dsl_str(self) -> str:
+        dsl_json_str = self.json(by_alias=True, exclude_none=True)
+        dsl_query_dict = dict(query=loads(dsl_json_str))
+        return dumps(dsl_query_dict)
```

Comment:
As discussed, let's embed calling this into the `create()` method (i.e. have the `create()` method take a `FluentSearch` object or `IndexSearchRequest` object itself, and translate it to a string inside the `create()` method), rather than relying on folks to remember to call it themselves.

Reply:
I tried using `FluentSearch`, but it didn't work due to a circular import error. Therefore, I opted to use the `IndexSearchRequest` object directly and translated it into string inside the `create()` method of the `DataProduct`.